### PR TITLE
feat(@wallet): Show lightbox when clicking collectible

### PR DESCRIPTION
**Pull request by Cuteivist**
Task #13808 

### What does the PR do

* Created new LoadingErrorComponent to display rectangle for error state
* Added loading state for collectible details view
* Added menu for collectible details view
* Extended ImageContextMenu to work with Gifs and Videos
* Extended image popup to utilise Image too instead of only AnimatedImage. There are some collectibles that cannot be displayed in AnimatedImage
* Added StatusVideoModal to display video as lightbox

### Affected areas

* Wallet / collectibles details view
* StatusQ popup components 

## Failed to load in lightbox

modal_failed_to_load

## Video

https://github.com/status-im/status-desktop/assets/11396062/bf0e93b3-a188-461f-b9c3-a6e09006811a

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml [hunk 1]
```diff
@@ -119,6 +129,24 @@ StatusRoundedComponent {
         visible: !root.isError && !root.isLoading
     }
 
+    MouseArea {
+        anchors.fill: parent
+        propagateComposedEvents: true
```
Comment (stefandunca) [inline ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml:134]: I can't tell what is the reason of enabling `propagateComposedEvents`. I would expect to have side effects of allowing click to propagate to parent while handing it ourself. This way we could have a double handling in different contexts.

**Review by stefandunca — APPROVED** [review]
I left a question inline.

File: ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml [hunk 2]
```diff
@@ -119,6 +129,24 @@ StatusRoundedComponent {
...
+        enabled: root.interactive && mediaLoader.visible && mediaLoader.item
+        cursorShape: Qt.PointingHandCursor
+        acceptedButtons: Qt.LeftButton | Qt.RightButton
+        onClicked: {
+            if (mouse.button == Qt.RightButton) {
+                root.openContextMenu(mediaLoader.item)
+                return
+            }
+
+            if (componentMediaType === StatusRoundedMedia.MediaType.Image) {
```
Comment (dlipicar) [inline ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml:144]: bummer, big videos would've been nice

Comment (dlipicar) [reply ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml:144]: do we have some followup task for that? :D

File: ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml [hunk 3]
```diff
@@ -80,6 +94,11 @@ StatusRoundedComponent {
         return StatusRoundedMedia.MediaType.Unknown
     }
 
+    signal imageClicked(var image, bool plain)
```
Comment (dlipicar) [inline ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml:97]: sounds like we could have 2 single, generic signals (mediaClicked, openMediaContextMenu) if we add the mediaType to the parameters.

File: ui/imports/shared/status/StatusImageModal.qml [hunk 4]
```diff
@@ -31,39 +32,81 @@ StatusDialog {
...
             onClicked: {
                 if (mouse.button === Qt.LeftButton)
                     root.close()
-                if (mouse.button === Qt.RightButton)
-                    Global.openMenu(imageContextMenu,
-                                    messageImage,
-                                    { imageSource: messageImage.source, url: root.url})
+                if (imageLoader.isError || imageLoader.isLoading || mouse.button !== Qt.RightButton)
+                    return
+                const isGif = (!root.plain && imageLoader.item && imageLoader.item.playing) || imageLoader.source.toLowerCase().endsWith(".gif")
```
Comment (dlipicar) [inline ui/imports/shared/status/StatusImageModal.qml:60]: hmm I don't think we should rely on the file extension to determine the media type.

File: ui/imports/shared/views/chat/ImageContextMenu.qml [hunk 5]
```diff
@@ -9,24 +9,25 @@ StatusMenu {
     property string imageSource
     property string domain
     property bool requireConfirmationOnOpen: false
+    property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
+    property bool isVideo: root.imageSource.toLowerCase().endsWith(".mp4")
```
Comment (dlipicar) [inline ui/imports/shared/views/chat/ImageContextMenu.qml:13]: 😭 could be `avi`, `mpg` or some random url that redirects to a video. We shouldn't rely on the file extension

**Comment by Khushboo-dev-cpp** [comment]
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=22617-263897&mode=design&t=3s5fO2jg8rAod8CL-4 wanted to confirm if have added the aspect ratio changes from design here?

File: ui/app/AppLayouts/Wallet/views/collectibles/CollectibleDetailView.qml [hunk 6]
```diff
@@ -14,6 +14,7 @@ import AppLayouts.Communities.panels 1.0
 import utils 1.0
 import shared.controls 1.0
 import shared.views 1.0
+import shared.views.chat 1.0
```
Comment (dlipicar) [inline ui/app/AppLayouts/Wallet/views/collectibles/CollectibleDetailView.qml:17]: this feels weird... those dependencies should probably be moved to a more generic location

File: ui/imports/shared/status/StatusVideoModal.qml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    padding: 0
+    background: null
+    standardButtons: Dialog.NoButton
+    closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutside
+
+    QtObject {
+        id: d
+
+        property int maxHeight: Global.applicationWindow.height - 80
+        property int maxWidth: Global.applicationWindow.width - 80
```
Comment (caybro) [inline ui/imports/shared/status/StatusVideoModal.qml:28]: `readonly` for both

File: ui/imports/shared/views/chat/ImageContextMenu.qml (hunk 5 above)
Comment (caybro) [reply ui/imports/shared/views/chat/ImageContextMenu.qml:13]: I agree with both points... but this is hard to do with remote URLs

File: ui/imports/shared/views/chat/ImageContextMenu.qml [hunk 8]
```diff
@@ -9,24 +9,25 @@ StatusMenu {
...
 
     QtObject {
         id: d
         readonly property bool isUnfurled: (!!url&&url!=="")
-        readonly property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
     }
 
     StatusAction {
-        text: d.isGif ? qsTr("Copy GIF") : qsTr("Copy image")
+        text: isGif ? qsTr("Copy GIF") : qsTr("Copy image")
```
Comment (caybro) [inline ui/imports/shared/views/chat/ImageContextMenu.qml:21]:
```suggestion
        text: root.isGif ? qsTr("Copy GIF") : qsTr("Copy image")
```

File: ui/imports/shared/views/chat/ImageContextMenu.qml [hunk 9]
```diff
@@ -9,24 +9,25 @@ StatusMenu {
...
-        enabled: !!root.imageSource
+        enabled: !!root.imageSource && !root.isVideo
         onTriggered: {
             Utils.copyImageToClipboardByUrl(root.imageSource)
         }
     }
 
     StatusAction {
-        text: d.isGif ? qsTr("Download GIF") : qsTr("Download image")
+        text: isGif ? qsTr("Download GIF") : isVideo ? qsTr("Download video") : qsTr("Download image")
```
Comment (caybro) [inline ui/imports/shared/views/chat/ImageContextMenu.qml:30]:
```suggestion
        text: root.isGif ? qsTr("Download GIF") : root.isVideo ? qsTr("Download video") : qsTr("Download image")
```

**Review by caybro — APPROVED** [review]
LGTM

just some minor things

**Pull request merged**